test/e2e/infraprovider/providers/kind/kind.go

```diff
@@ -359,6 +361,85 @@ func (c *contextKind) getAttachedNetworks() (api.Networks, error) {
 	return attachedNetworks, nil
 }
 
+func (c *contextKind) SetupUnderlay(f *framework.Framework, underlay api.Underlay) error {
+	if underlay.LogicalNetworkName == "" {
+		return fmt.Errorf("underlay logical network name must be set")
+	}
+
+	if underlay.PhysicalNetworkName == "" {
+		underlay.PhysicalNetworkName = "underlay"
+	}
+
+	if underlay.BridgeName == "" {
+		underlay.BridgeName = secondaryBridge
+	}
+
+	const (
+		ovsNodeLabel = "app=ovnkube-node"
+	)
+
+	ovsPodList, err := f.ClientSet.CoreV1().Pods(deploymentconfig.Get().OVNKubernetesNamespace()).List(
+		context.Background(),
+		metav1.ListOptions{LabelSelector: ovsNodeLabel},
+	)
+	if err != nil {
+		return fmt.Errorf("failed to list OVS pods: %w", err)
+	}
+
+	if len(ovsPodList.Items) == 0 {
+		return fmt.Errorf("no OVS pods found with label %s", ovsNodeLabel)
+	}
+
+	for _, ovsPod := range ovsPodList.Items {
+		if underlay.BridgeName != deploymentconfig.Get().ExternalBridgeName() {
+			underlayInterface, err := getNetworkInterface(ovsPod.Spec.NodeName, underlay.PhysicalNetworkName)
+			if err != nil {
+				return fmt.Errorf("failed to get underlay interface for network %s: %w", underlay.PhysicalNetworkName, err)
+			}
+			c.AddCleanUpFn(func() error {
+				if err := removeOVSBridge(ovsPod.Namespace, ovsPod.Name, underlay.BridgeName); err != nil {
+					return err
+				}
+				return nil
+			})
+			if err := addOVSBridge(ovsPod.Namespace, ovsPod.Name, underlay.BridgeName); err != nil {
+				return err
+			}
+
+			if err := ovsAttachPortToBridge(ovsPod.Namespace, ovsPod.Name, underlay.BridgeName, underlayInterface.InfName); err != nil {
+				return err
+			}
+			if underlay.VlanID > 0 {
+				if err := ovsEnableVLANAccessPort(ovsPod.Namespace, ovsPod.Name, underlay.BridgeName, underlayInterface.InfName, underlay.VlanID); err != nil {
+					return err
+				}
+			}
+		}
+		c.AddCleanUpFn(func() error {
```

Comment:
cleanup before success  of config'ing the bridge again - any reason?

Reply:
> cleanup before success  of config'ing the bridge again - any reason?

I understand that it feels like we should rollback to defaults just if we change them but sometimes
when developing you do really change things and still fail at modification and stuff do not go back to normal if the Defer is not before that, those functions are idempotent so no problem if we call them 
even if they are not needed.

Reply:
ah i didnt know they idempotent thanks!